dev/TeachingTip/TeachingTip.cpp

```diff
@@ -598,111 +597,63 @@ void TeachingTip::UpdateSizeBasedTemplateSettings()
 
 void TeachingTip::UpdateButtonsState()
 {
-    hstring actionText = ActionButtonText();
-    hstring closeText = CloseButtonText();
-    switch (CloseButtonKind())
+    auto actionContent = ActionButtonContent();
+    auto closeContent = CloseButtonContent();
+    bool isLightDismiss = IsLightDismissEnabled();
+    if (actionContent && closeContent)
     {
-    case winrt::TeachingTipCloseButtonKind::Auto:
-        if (actionText.size() > 0 && closeText.size() > 0)
-        {
-            winrt::VisualStateManager::GoToState(*this, L"BothButtonsVisible"sv, false);
-            winrt::VisualStateManager::GoToState(*this, L"FooterCloseButton"sv, false);
-        }
-        else if (actionText.size() > 0)
-        {
-            winrt::VisualStateManager::GoToState(*this, L"ActionButtonVisible"sv, false);
-            winrt::VisualStateManager::GoToState(*this, L"HeaderCloseButton"sv, false);
-        }
```

Comment:
If this was just a rename, why did the VSM logic change? (I'll admit I didn't really have a handle on it before either so it's hard for me to grok this delta)

Reply:
I'm assuming it has to do with changing the string to an object? It would have been much easier to understand without all the mechanical renames in the same PR.

Reply:
This has a slightly larger delta because of the removal of the CloseButtonKind api. Some of the logic for that API has moved here in its Auto form.

---
In reply to: [265702311](https://github.com/Microsoft/microsoft-ui-xaml/pull/409#discussion_r265702311) [](ancestors = 265702311)